calico/felix/plugins/fiptgenerator.py

```diff
@@ -576,13 +578,29 @@ def drop_rules(self, ip_version, chain_name, rule_spec=None, comment=None,
                 "Invalid comment %r" % comment
             comment_str = '-m comment --comment "%s"' % comment
 
-        parts = [p for p in [ipt_action,
-                             chain_name,
-                             rule_spec,
-                             '--jump DROP',
-                             comment_str]
-                 if p is not None]
-        return [' '.join(parts)]
+        rules = []
+
+        if self.ACTION_ON_DROP.startswith("LOG-"):
+            # log-and-accept, log-and-drop.
+            log_spec = '--jump LOG --log-prefix "CalicoDrop" --log-level 4'
+            log_rule = " ".join(
+                [p for p in [ipt_action, chain_name, rule_spec, log_spec,
+                             comment_str] if p is not None]
+            )
+            rules.append(log_rule)
+
+        if self.ACTION_ON_DROP.endswith("ACCEPT"):
+            action_spec = "--jump ACCEPT"
```

Comment:
There should be a comment in every overridden rule that says that it's overridden.  This is a nasty option to leave on unintentionally: anything we can do to make it more obvious has to be a good thing.